sql/core/src/main/scala/org/apache/spark/sql/catalog/Catalog.scala

```diff
@@ -175,13 +175,37 @@ abstract class Catalog {
       options: Map[String, String]): DataFrame
 
   /**
-   * Drops the temporary table with the given table name in the catalog.
-   * If the table has been cached before, then it will also be uncached.
+   * Drops the temporary view with the given view name in the catalog.
+   * If the view has been cached before, then it will also be uncached.
    *
-   * @param tableName the name of the table to be dropped.
+   * @param viewName the name of the view to be dropped.
    * @since 2.0.0
    */
-  def dropTempTable(tableName: String): Unit
+  def dropTempView(viewName: String): Unit
+
+
+  /**
+   * Creates a temporary view in the catalog with the given [[DataFrame]].
+   *
+   * @param viewName the name of the view to be created
+   * @df the DataFrame used to create the view.
+   *
+   * @throws AnalysisException if the view name already exists
+   * @since 2.0.0
+   */
+  @throws[AnalysisException]
+  def createTempView(viewName: String, df: DataFrame): Unit
```

Comment:
As it stands right now, Catalog.scala doesn't have any other interface for creating tables, so I'd prefer not having them for views for now.